codec/AudioReceivePipeline.cpp

```diff
@@ -481,13 +491,23 @@ void AudioReceivePipeline::process(const uint64_t timestamp)
 
         if (isDiscardedPacket(*packet))
         {
-            decodedSamples = std::max(480u, _metrics.receivedRtpCyclesPerPacket);
-            _decoder.onUnusedPacketReceived(extendedSequenceNumber);
+            if (!isG711(*packet))
+            {
+                decodedSamples = std::max(480u, _metrics.receivedRtpCyclesPerPacket);
+                _decoder.onUnusedPacketReceived(extendedSequenceNumber);
+            }
             std::memset(audioData, 0, decodedSamples);
         }
         else
         {
-            decodedSamples = decodePacket(extendedSequenceNumber, timestamp, *packet, audioData);
+            if (!isG711(*packet))
+            {
+                decodedSamples = decodePacket(extendedSequenceNumber, timestamp, *packet, audioData);
+            }
+            else
+            {
+                decodedSamples = decodeG711(extendedSequenceNumber, timestamp, *packet, audioData);
+            }
```

Comment:
Can abstract all this complexity by creating a common AudioDecoder interface for G711 and Opus?

AudioReceivePipeline could have a pointer to a decoder implementation that could keep all the decoder state.

I think it would simplify unit tests too

Like:
```
class AudioDecoder
{
    virtual void onUnusedPacketReceived(uint32_t extendedSequenceNumber) = 0;
    virtual  size_t decodePacket(uint32_t extendedSequenceNumber,
        uint64_t timestamp,
        const memory::Packet& packet,
        int16_t* audioData); = 0;
}
```